flink-staging/flink-streaming/flink-storm-compatibility/src/main/java/org/apache/flink/stormcompatibility/api/FlinkClient.java

```
@@ -0,0 +1,362 @@
+		}
+		
+		ActorRef jobManager;
+		try {
+			jobManager = JobManager.getJobManagerRemoteReference(new InetSocketAddress(this.jobManagerHost,
+				this.jobManagerPort), actorSystem, AkkaUtils.getLookupTimeout(configuration));
+			
+			if(options != null) {
+				try {
+					Thread.sleep(1000 * options.get_wait_secs());
+				} catch(final InterruptedException e) {
+					e.printStackTrace();
+				}
+			}
+			
+			final Future<Object> response = Patterns.ask(jobManager, new CancelJob(this.topologyNameToJobId.get(name)),
+				new Timeout(askTimeout));
+			
+			try {
+				Await.result(response, askTimeout);
+			} catch(final Exception e) {
+				throw new RuntimeException("Killing topology " + name + " with Flink job ID "
+					+ this.topologyNameToJobId.get(name) + " failed.", e);
+			}
+			
+			this.topologyNameToJobId.remove(name);
+			this.topologies.remove(name);
+		} catch(final IOException e) {
+			throw new RuntimeException("Could not connect to Flink JobManager with address " + this.jobManagerHost
+				+ ":" + this.jobManagerPort, e);
+		}
+	}
+	
+	// not sure if the following methods can be supported or not...
+	
+	// public void activate(final String name) throws NotAliveException {}
+	
+	// public void deactivate(final String name) throws NotAliveException {}
+	
+	// public void rebalance(final String name, final RebalanceOptions options) throws NotAliveException,
+	// InvalidTopologyException {}
+	
+	// public String beginFileUpload() {
+	// return null;
+	// }
+	
+	// public void uploadChunk(final String location, final ByteBuffer chunk) {}
+	
+	// public void finishFileUpload(final String location) {}
+	
+	// public String beginFileDownload(final String file) {
+	// return null;
+	// }
+	
+	// public ByteBuffer downloadChunk(final String id) {
+	// return null;
+	// }
+	
+	// public void send_downloadChunk(final String id) {}
+	
+	// public String getNimbusConf() {
+	// return null;
+	// }
```

I would like to keep it or to change it to throw UnsupportedOperationException(). This ensures, that the user is aware what operations are supported (or not supported) by the compability layer.